SteakBot.Core/Modules/QuoteModule.cs

```diff
@@ -11,62 +11,251 @@ namespace SteakBot.Core.Modules
                 {
-                    referredChannel = channel;
-                    lines.RemoveAt(0);
+                    channel = tmpChannel;
+                    return true;
                 }
             }
 
-            var embedDescription = new StringBuilder();
-            var embedAuthor = new EmbedAuthorBuilder();
+            channel = null;
+            return false;
+        }
 
-            var timestamp = string.Empty;
+        private static bool TryGetChannel(SocketGuild guild, ulong channelId, out SocketTextChannel channel)
+        {
+            foreach (var socketGuildChannel in guild.Channels)
+            {
+                if (socketGuildChannel is SocketTextChannel tmpChannel && tmpChannel.Id == channelId)
+                {
+                    channel = tmpChannel;
+                    return true;
+                }
+            }
 
-            foreach (var line in lines)
+            channel = null;
+            return false;
+        }
+
+        private bool TryGetGuild(ulong guildId, out SocketGuild guild)
+        {
+            foreach (var clientGuild in Context.Client.Guilds)
             {
-                var isAuthorLine = false;
-                foreach (var userList in usersByName)
+                if (clientGuild.Id == guildId)
                 {
-                    if (line.StartsWith(userList.Key))
-                    {
-                        embedAuthor = new EmbedAuthorBuilder
-                        {
-                            Name = userList.Key,
-                            IconUrl = userList.Value.Count == 1 ? userList.Value.First().GetAvatarUrl() : null
-                        };
+                    guild = clientGuild;
+                    return true;
+                }
+            }
 
-                        isAuthorLine = true;
+            guild = null;
+            return false;
+        }
 
-                        var trim = line.Substring(userList.Key.Length);
-                        timestamp = trim.Trim();
+        private async Task<List<IMessage>> GetMessageList(SocketTextChannel channel, ulong firstMessageId, ulong lastMessageId)
+        {
+            var firstMessage = await channel.GetMessageAsync(firstMessageId);
+
+            var downloadedMessages = channel.GetMessagesAsync(firstMessageId, Direction.After, 25);
+            var firstPage = (await downloadedMessages.First()).ToArray();
+
+            var messages = new List<IMessage>
+            {
+                firstMessage
+            };
+
+            for (var i = firstPage.Length - 1; i >= 0; i--)
+            {
+                if (firstPage[i].Author == firstMessage.Author)
+                {
+                    messages.Add(firstPage[i]);
+                }
 
-                        break;
-                    }
+                if (firstPage[i].Id == lastMessageId)
+                {
+                    break;
                 }
+            }
```

Comment:
Maybe add this whole thing in a while loop until last message id has been found?

Reply:
The idea of downloading 25 messages instead of the default 100 was to make a sensible balance between downloading as few messages as possible in order to speed things up and downloading enough to actually find it.
A quote spanning 25 messages seems like a valid limit to me.
Also if we make this loop and download more messages, then we need to add a retry count on that, which effectively is the same as increasing the download count.